synapse/tools/layer/dump.py

```diff
@@ -0,0 +1,170 @@
+            mesg = f'No edits to export starting from offset ({soffs})'
+            raise s_exc.BadArg(mesg=mesg)
+
+        finished = False
+
+        genr = layer.syncNodeEdits2(soffs, wait=False)
+
+        nodeiter = aiter(genr)
+
+        while not finished:
+
+            kwargs = {}
+
+            if s_common.majmin >= (3, 12): # pragma: no cover
+                kwargs['delete_on_close'] = False
+
+            with tempfile.NamedTemporaryFile(dir=opts.outdir, delete=False, **kwargs) as fd:
+
+                try:
+                    # Pull the first edit so we can get the starting offset
+                    first = await anext(nodeiter)
+                except StopAsyncIteration: # pragma: no cover
+                    break
+
+                soffs = first[0]
+
+                count = 1
+
+                # Write header to file
+                fd.write(s_msgpack.en((
+                    'init',
+                    {
+                        'hdrvers': 1,
+                        'celliden': celliden,
+                        'cellvers': cellvers,
+                        'layriden': opts.iden,
+                        'offset': soffs,
+                        'chunksize': opts.chunksize,
+                        'tick': s_common.now(),
+                    }
+                )))
+
+                # Now write the first edit that we already pulled
+                fd.write(s_msgpack.en(('edit', first)))
+
+                async for nodeedit in nodeiter:
+
+                    # Write individual edits to file
+                    fd.write(s_msgpack.en(('edit', nodeedit)))
+
+                    eoffs = nodeedit[0]
+
+                    count += 1
+
+                    if count % opts.chunksize == 0:
+                        break
+
+                else:
+                    finished = True
+
+                # Write footer to file
+                fd.write(s_msgpack.en(('fini', {
+                    'offset': eoffs,
+                    'tock': s_common.now(),
+                })))
+
+                tmpname = fd.name
+
+            path = os.path.join(opts.outdir, f'{celliden}.{opts.iden}.{soffs}-{eoffs}.nodeedits')
+            os.rename(tmpname, path)
+            outp.printf(f'Wrote layer node edits {soffs}-{eoffs} to {path}.')
+
+            # Update start offset for next loop
+            soffs = eoffs + 1
+
+    # Save state file
+    state['offset:next'] = eoffs + 1
+    s_common.yamlsave(state, statefile)
+
+    return 0
+
+async def main(argv, outp=s_output.stdout):
+
+    pars = argparse.ArgumentParser(prog='layer.dump', description=descr)
+    pars.add_argument('--url', default='cell:///vertex/storage', help='The telepath URL of the Synapse service.')
+    pars.add_argument('--offset', default=None, type=int, help='The starting offset of the node edits to export.')
+    pars.add_argument('--chunksize', default=100_000, type=int, help='The number of node edits to store in a single file.')
```

Comment:
We should be able and willing to dump/load any number of edits into a stream file. Why create a synthetic limit?